### PR TITLE
[WIP] Feature flags

### PR DESCRIPTION
**Pull request by russellb**
for issues #213 and #214

TODO:
- [x] Add Ingress for the feature flag proxy
- [ ] Update UI to disable multi-zone support based on feature flag status
- [ ] Move the apiserver back to polling an https URL for feature flags after issue #231 is resolved
- [ ] Document feature flags properly in docs/feature-flags.md

### CODE REVIEW

File: deploy/base/goff-proxy/kustomization.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+generatorOptions:
```
Comment (dave-tucker) [inline deploy/base/goff-proxy/kustomization.yaml:3]: I've just been through removing this in #229 - disableNameSuffixHash - makes the configmap immutable.
That means we'd have to delete/re-create it to pick up changes which I don't think is the behaviour we want.

File: internal/handlers/zone.go [hunk 2]
```diff
@@ -33,6 +34,17 @@ const (
...
 	ctx := c.Request.Context()
+
+	user := ffuser.NewUser(c.GetString(gin.AuthUserKey))
+	multiZoneEnabled, err := ffclient.BoolVariation("multi-zone", user, false)
+	if err != nil {
+		c.JSON(http.StatusInternalServerError, models.ApiError{Error: err.Error()})
+		return
+	}
+	if !multiZoneEnabled {
+		c.JSON(http.StatusForbidden, models.ApiError{Error: "multi-zone support is disabled"})
```
Comment (dave-tucker) [inline internal/handlers/zone.go:45]: http.StatusMethodNotAllowed would be more accurate

File: deploy/base/goff-proxy/kustomization.yaml (hunk 1 above)
Comment (russellb) [reply deploy/base/goff-proxy/kustomization.yaml:3]: ah, ok, yeah I'll remove it. It's just from copy/paste.

**Comment by russellb** [comment]
I'll squash all of these random commits once it's no longer a WIP.

**Comment by russellb** [comment]
I'm going to close this WIP PR in favor of introducing feature flags in smaller stages through multiple PRs.

**Pull request closed without merging**